### PR TITLE
airbyte-ci: use the base image to build java connectors

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/11151

We started publishing base image for our java connectors in https://github.com/airbytehq/airbyte/pull/49831 .
To make this base image used we should enable java connectors to express the base image they want to use in `metadata.yaml` like we do for java connectors.
This PR updates `airbyte-ci` to build java connectors from the base image declared in `metadata.yaml`.
It also marks the original build code paths as deprecated.

## Connector testing
* I successfully ran [`source-postgres` tests on this branch](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/augustin_12-17-airbyte-ci_use_the_base_image_to_build_java_connectors/1734429077/f6bebd4eae42420573f8886fa2679018527bc8bf/source-postgres/3.6.25/output.html). 🎉 
* I [pre-released a `source-postgres` version](https://hub.docker.com/layers/airbyte/source-postgres/3.6.25-dev.f6bebd4eae/images/sha256-3fe1ad38e2d37e50420488bc64d2b7c00836ff881cc4ade8c3831f42988e7064) if maintainers of this connectors want to test it out on Airbyte Cloud.

### CODE REVIEW

**Comment by alafanechere** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/49832?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#49832** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/49832?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/49832?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#49831** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/49831?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py [hunk 1]
```diff
@@ -388,3 +390,15 @@ async def raise_if_not_user(container: Container, expected_user: str) -> None:
...
+def deprecated(reason: str) -> Callable:
+    def decorator(func: Callable) -> Callable:
+        @functools.wraps(func)
+        def wrapper(*args: Any, **kwargs: Any) -> Any:
+            warnings.warn(f"{func.__name__} is deprecated: {reason}", DeprecationWarning, stacklevel=2)
+            return func(*args, **kwargs)
+
+        return wrapper
+
+    return decorator
```
Comment (postamar) [inline airbyte-ci/connectors/pipelines/pipelines/helpers/utils.py:404]: til!

File: airbyte-ci/connectors/pipelines/pipelines/dagger/containers/java.py [hunk 2]
```diff
@@ -158,22 +161,31 @@ async def with_airbyte_java_connector(context: ConnectorContext, connector_java_
...
+            f"Please declare a base image address in the connector metadata.yaml file (connectorBuildOptions.baseImage)."
+        )
+        base = with_integration_base_java(context, build_platform).with_entrypoint(["/airbyte/base.sh"])
 
     connector_container = (
         base.with_workdir("/airbyte")
         .with_env_variable("APPLICATION", application)
         .with_mounted_directory("built_artifacts", build_stage.directory("/airbyte"))
         .with_exec(sh_dash_c(["mv built_artifacts/* ."]))
-        .with_entrypoint(entrypoint)
```
Comment (postamar) [inline airbyte-ci/connectors/pipelines/pipelines/dagger/containers/java.py:177]: just checking: this is safe to remove? I don't actually know

**Review by postamar — APPROVED** [review]
Nice. I'm happy if the CI/CD is happy!

File: airbyte-ci/connectors/pipelines/pipelines/dagger/containers/java.py (hunk 2 above)
Comment (alafanechere) [reply airbyte-ci/connectors/pipelines/pipelines/dagger/containers/java.py:177]: Yes it it, I set it in the base image. And if the base image code path is not taken I added the entrypoint explicitely on line 177 and 183

**Pull request merged**